makerspace/views.py

```diff
@@ -0,0 +1,58 @@
+from django.http import Http404
+from django.views.generic import ListView, DetailView, UpdateView, CreateView, DeleteView, TemplateView
+from makerspace.models import Tool
+from django.contrib.auth.mixins import PermissionRequiredMixin
+from django.urls import reverse_lazy
+from web.templatetags.permission_tags import has_any_tool_permission
+
+
+class ToolView(DetailView):
+    model = Tool
+    template_name = 'tools/tool.html'
+    context_object_name = 'tool'
+
+
+class ToolsView(ListView):
+    model = Tool
+    template_name = 'tools/tools.html'
+    context_object_name = 'tools_list'
+
+
+class AdminToolView(TemplateView):
+    template_name = 'tools/admin_tool.html'
+
+    def get_context_data(self, **kwargs):
+        if not has_any_tool_permission(self.request.user) and not self.request.user.is_superuser:
```

Comment:
The explicit check for superuser can be removed. A superuser has all permissions.